web/frontend/src/components/navigation/plugins/render.ts

```diff
@@ -0,0 +1,202 @@
+import * as THREE from 'three';
+import { injectPlugin, useFrame, useRender, useThrelte } from '@threlte/core';
+import { MercatorCoordinate, type LngLat, LngLatBounds } from 'maplibre-gl';
+import { map, cameraMatrix, mapSize, view } from '../stores';
+
+const { clamp } = THREE.MathUtils;
+
+const renderTarget = new THREE.WebGLRenderTarget(0, 0, { format: THREE.RGBAFormat });
```

Comment:
Same question as above with all these module-level mutable singletons

Reply:
Yeah, these are things that we want cached due to the alternative being instantiating them in a `requestAnimationFrame`. Your question about cleanup is also legit - many of these THREE classes cache data on the GPU (usually buffers for geometric or texture data) and have `.dispose()` methods that need to be manually called when unmounting these components. It's for this reason that I'm switch us to use Threlte everywhere - the Threlte components that are analogous to THREE classes automatically dispose when unmounted. 

Unfortunately here this was a complex and uncommon rendering case that required a bit of render hacking. Luckily none of these THREE classes require manual disposal, so everything should be completely dereferenced an garbage collected on unmount.

Reply:
Oop! Actually there is one (`THREE.ShaderMaterial`). Will add cleanup. Thanks for the callout.